src/Http/Http.Abstractions/test/MapPathMiddlewareTests.cs

```diff
@@ -95,6 +95,34 @@ public void PathMatchAction_BranchTaken(string matchPath, string basePath, strin
             Assert.Equal(requestPath.Substring(matchPath.Length), context.Items["test.Path"]);
         }
 
+        [Theory]
+        [InlineData("/foo", "", "/foo")]
+        [InlineData("/foo", "", "/foo/")]
+        [InlineData("/foo", "/Bar", "/foo")]
+        [InlineData("/foo", "/Bar", "/foo/cho")]
+        [InlineData("/foo", "/Bar", "/foo/cho/")]
+        [InlineData("/foo/cho", "/Bar", "/foo/cho")]
+        [InlineData("/foo/cho", "/Bar", "/foo/cho/do")]
+        [InlineData("/foo", "", "/Foo")]
+        [InlineData("/foo", "", "/Foo/")]
+        [InlineData("/foo", "/Bar", "/Foo")]
+        [InlineData("/foo", "/Bar", "/Foo/Cho")]
+        [InlineData("/foo", "/Bar", "/Foo/Cho/")]
+        [InlineData("/foo/cho", "/Bar", "/Foo/Cho")]
+        [InlineData("/foo/cho", "/Bar", "/Foo/Cho/do")]
+        public void PathMatchAction_BranchTaken_WithoutRemoveMatchedPathSegment(string matchPath, string basePath, string requestPath)
+        {
+            HttpContext context = CreateRequest(basePath, requestPath);
+            var builder = new ApplicationBuilder(serviceProvider: null);
+            builder.Map(matchPath, false, subBuilder => subBuilder.Run(Success));
+            var app = builder.Build();
+            app.Invoke(context).Wait();
```

Comment:
Why are these tests doing .Wait instead of being async. Do they all do that @Tratcher ? If yes, can you file an issue to fix this?

Reply:
I blame me [6 years ago](https://github.com/aspnet/AspNetCore/blame/f6a2fe61a9a544899090fe21cb75250052a22640/src/Http/Http.Abstractions/test/MapPathMiddlewareTests.cs#L63). Amusingly the Waits were added when porting from Katana, [those tests](https://github.com/aspnet/AspNetKatana/blame/9a94c0f1b871549a7f6ff8ecf297548b6d46c8fa/tests/Microsoft.Owin.Mapping.Tests/MapPathMiddlewareTests.cs#L67) didn't wait at all :grin:.
Filed https://github.com/aspnet/AspNetCore/issues/15374

Reply:
I can fix this. Should I fix all tests in this PR or another one?